### ISSUE TITLE
Trying to downgradeComponent an Angular component into a root angular.js view

### ISSUE THREAD

**Issue by httpete** [ui-router/angular-hybrid#72]
I have a three tiered route setup, global.nav.feature.  I replaced uirouter/angularjs with the uirouter/hybrid and it is able to route to an Angular 5 component fine. But if I try to use downgradeComponent to get Angular components in my "global" component route (an angular.js view) it dies on any "*ngIf*. It also doesn't like any nested Angular components in my downgraded Components. Is this just not possible?

Error: StaticInjectorError(RootModule)[NgIf -> ViewContainerRef]: 
  StaticInjectorError(Platform: core)[NgIf -> ViewContainerRef]: 
    NullInjectorError: No provider for ViewContainerRef!
    at _NullInjector.get (core.js?2c1c:994)

**Comment by christopherthielen** [comment]
I'm not sure why this would happen.  Is the downgraded component loaded with root NgModule or lazy?  Let's say I had an [angular-hybrid app to start from](https://github.com/ui-router/sample-app-angular-hybrid), how would I re-create this?  Just downgrade a component and put it in the root angularjs view?

**Comment by christopherthielen** [comment]
This seems to work: https://github.com/ui-router/sample-app-angular-hybrid/compare/53f438e2828e11c0730508e989d7d04f154f9bd8...downgraded-component

**Comment by httpete** [comment]
Chris thank you for your reply - the issue was on my side but I will detail the fix here so others can avoid a headache.

If you are in the market for angular-hybrid you probably have a large app. I had a hand made webpack.config.js which worked fine for angularjs, but hit problems with Angular. In my use case I build an internal library which I develop via npm link to develop locally (symlinked via file:../my/package/dist in package.json)  , I use npm 5.3 in order to do this.

After seeing several people on posts complain, I narrowed it down to this webpack config.

`        resolve: {
            extensions: ['.ts', '.js'],
            modules: [
                "./node_modules",
                "./node_modules"
            ],
            symlinks: false,
            mainFields: [
                "browser",
                "module",
                "main"
            ]
        },`

Which I derived from an ejected angular-cli config. With symlinked npm packages, Angular gets module resolution messed up and with this it works. Thank you @christopherthielen  for a fantastic solution.
